### PR TITLE
Fix web alerts external/internal link and notification badge for AB#15899.

### PR DESCRIPTION
**Pull request by BrianMaki**
# Fixes [AB#15899](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15899)

## Description

- When new notifications are returned make sure last login date is in UTC when comparing to PHSA scheduled date 
- DateWrapper only requires either 'isUtc' or 'hasTime' but not both
- When external link is returned in web alert, make sure to use window open as it is not an internal link and cannot be used with router push.
- When internal link is returned in web alert, make sure it is valid internal route and can navigated to via router push. If it invalid route, then an error is logged for the time being as this needs to be address in another story @ninosamson @ytqsl 

**Notification Badge:**

Screenshot 2023-08-04 at 12 08 53 AM

**Notification Center:**

Screenshot 2023-08-04 at 12 14 24 AM

**Notification Center UI Functional Test:**

Screenshot 2023-08-03 at 11 29 25 PM

**Notification Center E2E Functional Test:**

Screenshot 2023-08-03 at 9 42 37 PM

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x] Not Required

## UI Changes

No

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/src/components/NotificationCentreComponent.vue [hunk 1]
```diff
@@ -63,10 +69,72 @@ function handleClickNotificationAction(notification: Notification): void {
...
     }
 }
 
+function isExternalUrl(url: string): boolean {
+    const currentDomain = window.location.hostname;
+    logger.debug(`Domain: ${currentDomain}`);
+
+    // Create a regular expression to extract the domain from the URL.
+    const domainRegex =
+        /^(?:https?:\/\/)?(?:[^@\n]+@)?(?:www\.)?([^:\/\n?]+)/im;
```
Comment (BronzBierd) [inline Apps/WebClient/src/ClientApp/src/components/NotificationCentreComponent.vue:101]: This is duplicated above and below, it should be moved to a const variable above.

Comment (BronzBierd) [inline Apps/WebClient/src/ClientApp/src/components/NotificationCentreComponent.vue:101]:
```suggestion
        /^(?:https:\/\/)?(?:[^@\n]+@)?(?:www\.)?([^:\/\n?]+)/im;
```

File: Apps/WebClient/src/ClientApp/src/components/NotificationCentreComponent.vue [hunk 2]
```diff
@@ -63,10 +69,72 @@ function handleClickNotificationAction(notification: Notification): void {
...
     } else {
-        router.push({ path: notification.actionUrl });
+        const isExternal = isExternalUrl(notification.actionUrl);
+        logger.debug(`is External: ${isExternal}`);
+
+        if (isExternal) {
+            // Open the external url in a new tab/window
+            window.open(notification.actionUrl, "_blank");
+        } else {
+            const internalRoute = stripInternalPath(notification.actionUrl);
```
Comment (BronzBierd) [inline Apps/WebClient/src/ClientApp/src/components/NotificationCentreComponent.vue:79]:
```suggestion
            const internalRoute = notification.actionUrl.replace(/(^\w+:|^)\/\/[^/]+/, "");
```

File: Apps/WebClient/src/ClientApp/src/components/NotificationCentreComponent.vue [hunk 3]
```diff
@@ -63,10 +69,72 @@ function handleClickNotificationAction(notification: Notification): void {
...
+        );
+
+        // Return the path segments starting from the domain index.
+        return `/${pathSegments.slice(domainIndex).join("/")}`;
+    }
+
+    // Return the url if the domain regex does not match (fallback behavior).
+    return url;
+}
+
```
Comment (BronzBierd) [inline Apps/WebClient/src/ClientApp/src/components/NotificationCentreComponent.vue:137]:
```suggestion
```

**Review by BronzBierd — DISMISSED** [review]

**Review by BronzBierd — APPROVED** [review]

**Review by ytqsl — APPROVED** [review]

**Review by BronzBierd — APPROVED** [review]

**Pull request merged**